### PR TITLE
Force rotation: Implements JWT endpoints in local authority API

### LINKED ISSUE #3896: Implement JWT Local Authority RPCs
Implement:
- Implement GetJwtAuthorityState
- implement PrepareJwtAuthority
- implement ActivateJwtAuthority
- Implement taintJwtAuthiority
- implement RevokeJWTAuthority

Depends on #3895, #3885

### PR DESCRIPTION
**Pull request by MarcosDY**
Implements JWT endpoints in local authority API

**Which issue this PR fixes**
fixes #3896

### CODE REVIEW

File: pkg/server/api/localauthority/v1/service_test.go [hunk 1]
```diff
@@ -43,6 +44,889 @@ var (
...
+						telemetry.Status:        "error",
+						telemetry.StatusCode:    "InvalidArgument",
+						telemetry.StatusMessage: "invalid authority ID: no authority ID provided",
+						telemetry.Type:          "audit",
+					},
+				},
+			},
+		},
+		{
+			name:        "no allow to revoke a prepared key",
```
Comment (amartinezfayo) [inline pkg/server/api/localauthority/v1/service_test.go:771]:
```suggestion
			name:        "not allow to revoke a prepared key",
```

**Review by amartinezfayo — COMMENTED** [review]
Thank you @MarcosDY, looks great!

File: pkg/server/api/localauthority/v1/service.go [hunk 2]
```diff
@@ -59,24 +58,172 @@ type Service struct {
...
+	case req.AuthorityId == "":
+		return nil, api.MakeErr(log, codes.InvalidArgument, "no authority ID provided", nil)
+
+	/// Only next local authority can be Activated
+	case req.AuthorityId != nextSlot.AuthorityID():
+		return nil, api.MakeErr(log, codes.InvalidArgument, "unexpected authority ID", nil)
+
+	// Only PREPARED local authorities can be Activated
+	case nextSlot.Status() != journal.Status_PREPARED:
+		return nil, api.MakeErr(log, codes.Internal, "only Prepared authorities can be activated", nil)
```
[the remark this answers is not in the document]
Comment (MarcosDY) [reply pkg/server/api/localauthority/v1/service.go:139]: not sure, since the caller already have the status in the request

**Review by amartinezfayo — APPROVED** [review]
Thank you @MarcosDY!

**Pull request merged**